ambry-router/src/main/java/com.github.ambry.router/GetOperation.java

```diff
@@ -141,29 +141,10 @@ boolean isOperationComplete() {
   abstract void handleResponse(ResponseInfo responseInfo);
 
   /**
-   * Process the given {@link ServerErrorCode} and set operation status accordingly.
-   * @param errorCode the {@link ServerErrorCode} to process.
+   * Abort operation by invoking any callbacks and updating futures with an exception.
+   * @param abortCause the exception that is the cause for the abort.
    */
-  void processServerError(ServerErrorCode errorCode) {
-    switch (errorCode) {
-      case Blob_Deleted:
-        logger.trace("Requested blob was deleted");
-        setOperationException(new RouterException("Server returned: " + errorCode, RouterErrorCode.BlobDeleted));
-        break;
-      case Blob_Expired:
-        logger.trace("Requested blob is expired");
-        setOperationException(new RouterException("Server returned: " + errorCode, RouterErrorCode.BlobExpired));
-        break;
-      case Blob_Not_Found:
-        logger.trace("Requested blob was not found on this server");
-        setOperationException(new RouterException("Server returned: " + errorCode, RouterErrorCode.BlobDoesNotExist));
-        break;
-      default:
-        setOperationException(
-            new RouterException("Server returned: " + errorCode, RouterErrorCode.UnexpectedInternalError));
-        break;
-    }
-  }
+  abstract void abort(Exception abortCause);
```

Comment:
Can you throw some light as to why we need this method. Also, what is the difference between this method and setOperationException